include/fiction/algorithms/simulation/sidb/check_equivalence_simulation_result.hpp

```diff
@@ -0,0 +1,89 @@
+//
+// Created by Jan Drewniok on 03.03.24.
+//
+
+#ifndef FICTION_CHECK_EQUAL_CDS_RESULT_HPP
+#define FICTION_CHECK_EQUAL_CDS_RESULT_HPP
+
+#include "fiction/algorithms/simulation/sidb/sidb_simulation_result.hpp"
+
+#include <algorithm>
+#include <cmath>
+#include <limits>
+
+namespace fiction
+{
+
+/**
+ * This function compares two SIDB simulation results for equivalence. Two results are considered equivalent if they
+ * have the same number of charge distributions and if each corresponding charge distribution has the same electrostatic
+ * potential energy and charge states for all cells.
+ *
+ * @tparam Lyt The layout type used in the simulation results.
+ * @param result1 The first SiDB simulation result to compare.
+ * @param result2 The second SiDB simulation result to compare.
+ * @return `true` if the two simulation results are equivalent, `false otherwise.
+ */
+template <typename Lyt>
+[[nodiscard]] bool check_equivalence_simulation_result(sidb_simulation_result<Lyt>& result1,
```

Comment:
Can we rename the function to `check_equivalence_of_simulation_results` or `check_simulation_results_for_equivalence`?